src/network/networkd-dhcp-common.c

```diff
@@ -279,6 +279,67 @@ int config_parse_dhcp6_mud_url(
                 void *data,
                 void *userdata) {
 
+        char ***l = data;
+        int r;
+
+        assert(l);
+        assert(lvalue);
+        assert(rvalue);
+
+        if (isempty(rvalue)) {
+                *l = strv_free(*l);
+                return 0;
+        }
+
+        for (;;) {
+                _cleanup_free_ char *w = NULL;
+
+                r = extract_first_word(&rvalue, &w, NULL, EXTRACT_CUNESCAPE|EXTRACT_UNQUOTE);
+                if (r == -ENOMEM)
+                        return log_oom();
+                if (r < 0) {
+                        log_syntax(unit, LOG_ERR, filename, line, r,
+                                   "Failed to split user classes option, ignoring: %s", rvalue);
+                        break;
+                }
+                if (r == 0)
+                        break;
+
+                if (ltype == AF_INET) {
+                        if (strlen(w) > 255) {
```

Comment:
maybe UINT8_MAX?

in other words: i think it would make sense to either spell out only 255, 65535, and so on as decimal values. Or to spell them out as 0xff and 0xffff instead as hexadecimal values, or as UINT8_MAX and UINT16_MAX instead, i.e. as macros. But please spell it out using the same style of the three everywhere. I'd personally use either the hex or the macro form, they make clearest what this is, with a slight preference for the macro form. But important is not so much which of the three formats you pick, but that you peak the same form for all size checks here. No you picked decimal for the dhcpv4 case but macro for dhcpv6 and that's just confusing...

Reply:
Yes `UINT8_MAX` Is nicer than 255.